web-client/public/js/grnstate.js

```diff
@@ -31,16 +32,20 @@ export const grnState = {
 
     newNetwork: false,
 
+    normalizationMax : null,
+    resetNormalizationMax: null,
+
     get network () {
         return currentNetwork;
     },
 
     set network (network) {
         currentNetwork = network;
+        this.resetNormalizationMax = max(network.positiveWeights.concat(network.negativeWeights));
```

Comment:
Whoa, after all that code in _graph.js_, this is what it boils down to? Shouldn't https://github.com/dondi/GRNsight/blob/beta/web-client/public/js/graph.js#L62-L72 be done to `allWeights` first?

(and yes, if you're thinking "but that needs `colorOptimal`" —then you have your next candidate to move into `grnState` 🙂)

Reply:
It would be understandable if we pushed `colorOptimal` to another phase, but in that case, leave a TODO comment here. You'd be amazed at how many times a developer might say "I'll get to that later" but they forget because other stuff comes up. Putting a comment here will help guide your future self.